pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -1670,52 +1709,41 @@ func (service *SecurityPolicyService) updatePeerExpressions(obj *v1alpha1.Securi
 	return totalCriteriaCount, totalExprCount, nil
 }
 
-func (service *SecurityPolicyService) buildShareName(nsxProjectName, groupName string) string {
-	nsxProjectShareName := util.GenerateTruncName(common.MaxNameLength, nsxProjectName, common.SharePrefix, fmt.Sprintf("group-%s", groupName), "", "")
-	return nsxProjectShareName
+func (service *SecurityPolicyService) buildShareName(nsxResourceID, groupName string) string {
+	nsxShareName := util.GenerateTruncName(common.MaxNameLength, fmt.Sprintf("%s-group-%s", nsxResourceID, groupName), "", common.ShareSuffix, "", "")
+	return nsxShareName
```

Comment:
This is because that the PR https://github.com/vmware-tanzu/nsx-operator/pull/643 refactored the ID/Name format for our Operator Resource, the resource type is no longer added as a prefix for a resource. 
Also, it's recommended that add other info as suffix instead of prefix. 
Hence, I changed the share resource to align the wenying's PR change.